sdk/dotnet/Pulumi/Config_Exceptions.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright 2016-2019, Pulumi Corporation
+
+using System;
+
+namespace Pulumi
+{
+    public partial class Config
+    {
+        /// <summary>
+        /// ConfigTypeException is used when a configuration value is of the wrong type.
+        /// </summary>
+        private class ConfigTypeException : RunException
```

Comment:
I don't think so tbh.  Honestly, the type could just go away.  All we're doing is throwing a RunException so that we'll print a nice message to the console and then quit.